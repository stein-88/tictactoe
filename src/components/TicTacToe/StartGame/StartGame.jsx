import React from 'react'
import virus from '@images/tictactoe.png'
import Btn from '@generics/Btn'
// import { v4 as uuidv4 } from 'uuid'
import schema from './data'

const StartGame = ({ changeToPage }) => {
	if (!schema || !changeToPage) return null
	const { titulo, players, btntitle } = schema
	return (
		<div className="row">
			<div className="col-lg-6 col-12">
				<p className="h1 pb-5">{titulo}</p>
				{players && players.length > 0
					&& players.map((player, ind) => {
						const { play, strong } = player
						return (
							<p
								key={`fdasfdsa${ind + 1}`}
								data-test="players"
								className={`h3 ${ind === 0 ? 'py-3' : ''.trim()}`}
							>
								{play}
								<strong className="text-danger">{strong}</strong>
							</p>
						)
					})}
			</div>
			<div className="col-lg-6 col-12">
				<section className="w-100 text-center">
					<section className="w-100 px-5 mb-3">
						<img className="w-100 img-fluid rounded" src={virus} alt={titulo} />
					</section>
					<Btn text={btntitle} onClick={() => changeToPage()} />
				</section>
			</div>
		</div>
	)
}

export default StartGame
