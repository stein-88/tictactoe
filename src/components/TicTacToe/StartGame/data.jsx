import { PLAYERS } from '@constants/config'

const [play1, play2] = PLAYERS
const schemastartgame = {
	titulo: 'Tic Tac Toe',
	btntitle: 'Start Game',
	players: [
		{
			play: 'Player One: ',
			strong: play1,
		},
		{
			play: 'Player Two: ',
			strong: play2,
		},
	],
}

export default schemastartgame
