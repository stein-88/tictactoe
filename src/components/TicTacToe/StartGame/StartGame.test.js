import React from 'react'
import renderer from 'react-test-renderer'
import StartGame from './StartGame'
import schema from './data'
const { players, titulo } = schema
const verify = [titulo]

describe('Test on StartGame Component', () => {
	const mockFunc = jest.fn()
	const compStartGame = renderer.create(<StartGame changeToPage={mockFunc} />)

	test('Should not render without props', () => {
		const compStartGame = renderer.create(<StartGame />)
		const jsonStartGame = compStartGame.toJSON()
		expect(jsonStartGame).toBeNull()
	})
	test('Should render with props and have 2players only', () => {
		const jsonStartGame = compStartGame.toJSON()
		const Result = compStartGame.root.findAllByProps({
			['data-test']: 'players'
		})
		expect(jsonStartGame).toMatchSnapshot()
		expect(Result).toHaveLength(players.length)
	})
	test('The component should have just 1 title h1', () => {
		const Result = compStartGame.root.findAllByProps({ className: 'h1 pb-5' })
		expect(Result).toHaveLength(verify.length)
	})
	test('Should have 1 buttom with aria-hidden', () => {
		const Result = compStartGame.root.findAllByProps({ className: 'btnstyle' })
		Result[0].props.onClick()
		expect(mockFunc).toHaveBeenCalled()
		expect(Result).toHaveLength(1)
	})
})
