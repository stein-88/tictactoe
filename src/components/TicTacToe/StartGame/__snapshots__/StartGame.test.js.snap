// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on StartGame Component Should render with props and have 2players only 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-6 col-12"
  >
    <p
      className="h1 pb-5"
    >
      Tic Tac Toe
    </p>
    <p
      className="h3 py-3"
      data-test="players"
    >
      Player One: 
      <strong
        className="text-danger"
      >
        X
      </strong>
    </p>
    <p
      className="h3 "
      data-test="players"
    >
      Player Two: 
      <strong
        className="text-danger"
      >
        O
      </strong>
    </p>
  </div>
  <div
    className="col-lg-6 col-12"
  >
    <section
      className="w-100 text-center"
    >
      <section
        className="w-100 px-5 mb-3"
      >
        <img
          alt="Tic Tac Toe"
          className="w-100 img-fluid rounded"
          src="test-file-stub"
        />
      </section>
      <span
        aria-hidden="true"
        className="btnstyle"
        onClick={[Function]}
      >
        Start Game
      </span>
    </section>
  </div>
</div>
`;
